app/authentication/auth.py

```diff
@@ -120,17 +120,17 @@ def requires_auth():
             decode_jwt_token(auth_token, api_key.secret)
         except TokenDecodeError:
             continue
-        except TokenExpiredError:
+        except TokenExpiredError as tee:
             try:
                 decoded_token = decode_token(auth_token)
             except PyJWTError:
                 continue
             current_app.logger.info(f'JWT: iat value was {decoded_token["iat"]} while server clock is {epoch_seconds()}')
             err_msg = "Error: Your system clock must be accurate to within 30 seconds"
-            raise AuthError(err_msg, 403, service_id=service.id, api_key_id=api_key.id)
+            raise AuthError(err_msg, 403, service_id=service.id, api_key_id=api_key.id) from tee
 
         _auth_with_api_key(api_key, service)
-        return
+        break
```

Comment:
Why change this return to a break?

Reply:
This was a Python linter warning, mentioning that `break` would be more canon to the PEP of `else` used with a conditional. I manually tested to see if that works with a `return` as well and it does, but it seems that `break` would be more semantically correct. There should be difference in the end with this change as to how I understand this.

Reply:
BTW, I have my linter normally set to `mypy` in VSCode so I don't see these normally but as I worked in a devcontainer and `mypy` wasn't configured, I started to see all of these Python linter warnings/errors. I will probably switch between the two linters more often to bring these small fixes for more canonical Python / best practices.